src/posts/lucky_1.4.1.cr

```diff
@@ -0,0 +1,182 @@
+
+    See the [UPGRADE_NOTES](https://github.com/luckyframework/lucky/blob/main/UPGRADE_NOTES.md#upgrading-from-130-to-140).
+
+    You can also view the changes using [LuckyDiff](https://luckydiff.com/?from=1.3.0&to=1.4.1).
+
+    ## Here's what's new
+
+    ### Rate Limiting
+
+    Lucky Actions now have a "Rate Limiting" feature that allows you to limit the number of calls to an action within
+    a specified period. To use this feature, you will just need to include the new `Lucky::RateLimit` module.
+
+    ```crystal
+    class Users::LivePingCheck < ApiAction
+      include Lucky::RateLimit
+
+      rate_limit to: 1, within: 1.minute
+
+      get "/users/live_ping_check" do
+        plain_text "ok"
+      end
+
+      private def rate_limit_identifier : String
+        "live-ping-check"
+      end
+    end
+    ```
+
+    As you can see, this will limit the number of requests to the endpoint to a max of 1 every 1 minute. It will use the
+    `rate_limit_identifier` key for the lookup which you can customize if you need to set the limit to different amounts
+    on a per user basis, for example.
+
+    ### No more bash script setups
+
+    In order for us to better support Windows, we needed to remove the bash scripts that were used for bootstrapping a
+    Lucky application. While using Powershell would have been an option, this also would have made the code in the LuckyCLI
+    much more difficult to maintain due to having different codebase structures for Windows and non-Windows.
+
+    Instead, we decided to just port all of the scripts straight to Crystal. This is now a lot more cross-platform compatible
+    since you will need Crystal on any machine you're using anyway. You can now generate a new Lucky app on Windows, and
+    bootstrap the app with `crystal script/setup.cr`.
+
+    ### Separate Read/Write databases
+
+    Avram now supports using separate read and write databases for queries. By default, this will continue to use your
+    single specified database.
+
+    ```crystal
+    abstract class BaseModel < Avram::Model
+      def self.read_database : Avram::Database.class
+        ReaderDB
+      end
+
+      def self.write_database : Avram::Database.class
+        WriterDB
+      end
+    end
+
+    # Uses the ReaderDB
+    user = UserQuery.find(1)
+
+    # Uses the WriterDB
+    SaveUser.update!(user, last_active_at: Time.utc)
+    ```
+
+    Currently this will only change the connection between doing queries and saving records through Save or Delete operations.
+    Handling tasks such as fallbacks, or retries based on latency from reading and writing are not supported.
+
+    ### Table Locking
+
+    In general, a table lock is a way to lock your table while you perform operations on it preventing other connections
+    from altering that table during that time. In Postgres, however, there's 8 different types of locking that can
+    allow for a lot more customization and handling when it comes to locking your table.
```

Comment:
How's this? Still getting what you were trying to say? I removed the "generic table locks bit" cut to the Postgres chase.

```suggestion
    Table locks control which actions can be performed on a table at the same time. Postgres has
    [eight different table locking modes](https://www.postgresql.org/docs/current/explicit-locking.html#LOCKING-TABLES) that
    allow you to specify the locking behavior. Avram now supports all eight.
```

Reply:
ah, yeah, I like that better. Thanks!